### PR TITLE
Add download package to jetpack.backup

### PR DESCRIPTION
**Pull request by zwarm**
Parent #13329 
The PR adds the `download` package to the `Jetpack.Backup` package to separate the download process from the daily backup views.

To test:
- Launch the app 
- Tap on Activity Log
- Tap on the more menu
- Tap the download backup option
- Ensure the backup download view is shown

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadListItemState.kt [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+package org.wordpress.android.ui.jetpack.backup.download
 
 import androidx.annotation.DrawableRes
 import org.wordpress.android.R
 import org.wordpress.android.ui.jetpack.BackupAvailableItemsProvider.BackupAvailableItemType
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadListItemState.ViewType.ACTION_BUTTON
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadListItemState.ViewType.CHECKBOX
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadListItemState.ViewType.DESCRIPTION
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadListItemState.ViewType.HEADER
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadListItemState.ViewType.SUBHEADER
```
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadListItemState.kt:10]: Minor (🔍 ): Probably this was automatically done due to the way your `Optimize imports` works, not sure, we can check it out if you like. In the meanwhile, can you revert this change if that wasn't intentional?

**Review by ParaskP7 — APPROVED** [review]
Good job @zwarm ! 🌟 

I have tested the change and checked on the updated `backup` package structure, it looks great!

I have left one minor import related comment, but approving the PR so that I don't block you.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadListItemState.kt (hunk 1 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadListItemState.kt:10]: This is changed in a future PR. so all is good.
